android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt

```diff
@@ -1,22 +1,612 @@
+				)
+			}
+
+
+		}
+
+	}
+}
+
+@Composable
+fun FilterDropDown(
+	modifier: Modifier = Modifier,
+	filterName: String,
+	dropDownItems: Array<Int>,
+	updateState: (Int) -> Unit,
+	selectedValue: Int,
+	width: Dp,
 ) {
-	Column(
-		modifier = modifier.fillMaxSize(1.0f),
-		horizontalAlignment = Alignment.CenterHorizontally,
-		verticalArrangement = Arrangement.Center,
+	var expanded by remember { mutableStateOf(false) }
+
+	Box(
+		modifier = modifier
+			.width(width)
+			.height(dimensionResource(id = R.dimen.l))
+			.fillMaxWidth(1.0f)
 	) {
-		Text(text = "Home Page")
+		Button(
+			modifier = Modifier
+				.fillMaxWidth(1.0f)
+				.fillMaxHeight(1.0f),
+			contentPadding = PaddingValues(dimensionResource(id = R.dimen.s), ZERO_DP),
+			colors = ButtonDefaults.buttonColors(
+				containerColor = secondaryButtonBackgroundColor,
+				contentColor = secondaryTextColor,
+			),
+			shape = RoundedCornerShape(
+				dimensionResource(id = R.dimen.s)
+			),
+			onClick = {
+				expanded = !expanded
+			},
+		) {
+			Row(
+				modifier = Modifier.fillMaxWidth(1.0f),
+				verticalAlignment = Alignment.CenterVertically,
+				horizontalArrangement = Arrangement.SpaceBetween,
+			) {
+				Text(
+					text = filterName,
+					style = filterTextFont
+				)
+
+				Icon(
+					painter = painterResource(
+						id =
+						if (expanded) R.drawable.arrow_drop_up_soild_gray_24
+						else R.drawable.arrow_drop_down_solid_gray_24
+					),
+					contentDescription = stringResource(
+						id =
+						if (expanded) R.string.drop_up_arrow
+						else R.string.drop_down_arrow
+					)
+				)
+			}
+		}
+		DropdownMenu(
+			modifier = modifier
+
+				.background(Color.White),
+			expanded = expanded,
+			onDismissRequest = { expanded = false },
+		) {
+			dropDownItems.forEach {
+				DropdownMenuItem(
+					modifier = Modifier,
+					text = {
+						Text(
+							text = stringResource(it),
+							style = filterTextFont,
+							color = if (it == selectedValue && stringResource(it) != "Clear") subletrPink else Color.Black
+						)
+					},
+					contentPadding = PaddingValues(dimensionResource(id = R.dimen.s), ZERO_DP),
+					onClick = {
+						updateState(it)
+						expanded = false
+					},
+				)
+			}
+		}
+	}
+}
+
+@Composable
+fun ButtonWithIcon(
+	modifier: Modifier = Modifier,
+	iconId: Int,
+	onClick: () -> Unit,
+	contentDescription: String,
+	colors: ButtonColors = ButtonDefaults.buttonColors(
+		containerColor = secondaryButtonBackgroundColor,
+		contentColor = Color.Black
+	),
+) {
+	SecondaryButton(onClick = onClick,
+		modifier = modifier
+			.wrapContentSize(align = Alignment.Center),
+		contentPadding = PaddingValues(ZERO_DP),
+		colors = colors,
+		content = {
+			Icon(
+				painter = painterResource(
+					id = iconId
+				),
+				contentDescription = contentDescription,
+			)
+		})
+}
+@Composable
+fun ViewSwitch(modifier: Modifier = Modifier) {
+	val isListView = remember { mutableStateOf(true) }
```

Comment:
I think this should be a param since we'll need it in the page and not only in the switch